core/store/src/main/java/uk/gov/gchq/gaffer/store/library/GraphLibrary.java

```diff
@@ -216,4 +221,24 @@ private void checkExisting(final String graphId, final byte[] schema, final Stor
             }
         }
     }
+
+    private void checkSchemaExists(final Schema schema) {
+        if (null != getSchema(schema.getId())) {
+            if (!JsonUtil.equals(getSchema(schema.getId()).toJson(false), schema.toJson(false))) {
+                throw new OverwritingException("schemaId " + schema.getId() + " already exists with a different schema:\n"
+                        + "existing schema:\n" + StringUtil.toString(getSchema(schema.getId()).toJson(false))
+                        + "\nnew schema:\n" + StringUtil.toString(schema.toJson(false)));
+            }
+        }
+    }
+
+    private void checkPropertiesExist(final StoreProperties properties) {
+        if (null != getProperties(properties.getId())) {
+            if (!getProperties(properties.getId()).equals(properties)) {
```

Comment:
I "think" you need to compare the properties within StoreProperties.

Reply:
I have updated this logic in GraphLibrary now